url89849c8c5b_timmolter_XChart_tgz-pJ8-org_knowm_xchart_standalone_Example1J8/src/org/knowm/xchart/internal/chartpart/PlotSurface_AxesChart.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright 2015-2016 Knowm Inc. (http://knowm.org) and contributors.
+ * Copyright 2011-2015 Xeiam LLC (http://xeiam.com) and contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.knowm.xchart.internal.chartpart;
+
+import java.awt.Graphics2D;
+import java.awt.Shape;
+import java.awt.geom.Line2D;
+import java.awt.geom.Rectangle2D;
+import java.util.List;
+
+import org.knowm.xchart.XYSeries;
+import org.knowm.xchart.internal.Series;
+import org.knowm.xchart.style.AxesChartStyler;
+import org.knowm.xchart.style.Styler;
+
+/**
+ * Draws the plot background, the plot border and the horizontal and vertical grid lines
+ *
+ * @author timmolter
+ */
+public class PlotSurface_AxesChart<ST extends Styler, S extends Series> extends PlotSurface_ {
+
+  private final AxesChartStyler stylerAxesChart;
+
+  /**
+   * Constructor
+   *
+   * @param chart
+   */
+  protected PlotSurface_AxesChart(Chart<AxesChartStyler, XYSeries> chart) {
+
+    super(chart);
+    this.stylerAxesChart = chart.getStyler();
+  }
+
+  @Override
+  public void paint(Graphics2D g) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 64-96, lines 100-133 into separate methods.